### PR TITLE
Rainy day notice WIP

### PR DESCRIPTION
**Pull request by bendelonlee**
Doesn't make sense to merge yet. No users as yet have telephones or a way for them to add their phone numbers.

 I want to see if my tests pass travis.

### CODE REVIEW

File: app/jobs/rainy_day_texter_job.rb [hunk 1]
```diff
@@ -0,0 +1,30 @@
+class RainyDayTexterJob < ApplicationJob
+  queue_as :default
+
+  def perform
+    users_with_phone_numbers.each do |user|
+      send_rainy_day_text(user)
+    end
+    RainyDayTexterJob.set(wait_until: early_next_morning).perform_later
```
Comment (asmolentzov) [inline app/jobs/rainy_day_texter_job.rb]: `early_next_morning` this is awesome
general shout-out for readability-of-code for this.

File: app/jobs/rainy_day_texter_job.rb [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  end
+
+  def early_next_morning
+    self.class.early_next_morning
+  end
+
+  def self.early_next_morning
+    Time.use_zone("Mountain Time (US & Canada)") { 1.day.from_now.beginning_of_day + 5.hours }
```
Comment (asmolentzov) [inline app/jobs/rainy_day_texter_job.rb]: So basically 5am yah? An interesting eventual refactor might be taking the user's time zone (based on garden zip code or something?) and using that? I have no idea how much extra work that might be though.

File: app/jobs/rainy_day_texter_job.rb [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+  end
+
+  def users_with_phone_numbers
+    User.where.not(telephone: nil)
+  end
+
+  def send_rainy_day_text(user, chance = 110)
+    MyTwillioClient.api.account.messages.create(
+      from: '+12028834286',
```
Comment (asmolentzov) [inline app/jobs/rainy_day_texter_job.rb]: What is this phone number?

File: config/initializers/twillio.rb [hunk 4]
```diff
@@ -0,0 +1,6 @@
+require 'twilio-ruby'
+
+account_sid = ENV['TWILLIO_ACCOUNT_SID']
+auth_token = ENV['TWILLIO_AUTH_TOKEN']
```
Comment (asmolentzov) [inline config/initializers/twillio.rb]: So we'll prob all need these set? Are these for the app (sort of like google oauth), or personal? Aka should we all sign up with Twilio for our own, or do you mind sharing?

File: app/jobs/rainy_day_texter_job.rb (hunk 3 above)
Comment (bendelonlee) [reply app/jobs/rainy_day_texter_job.rb]: That phone number is related to my Twillio account

File: config/initializers/twillio.rb (hunk 4 above)
Comment (bendelonlee) [reply config/initializers/twillio.rb]: Right now they're not needed to pass any tests. Unsure of how to test twillio at the moment. You can find them in our Heroku console.

File: Gemfile [hunk 5]
```diff
@@ -59,6 +61,7 @@ group :development do
   # Access an IRB console on exception pages or by using <%= console %> anywhere in the code.
   gem 'web-console', '>= 3.3.0'
   gem 'listen', '>= 3.0.5', '< 3.2'
+  gem 'sinatra'
```
Comment (le3ah) [inline Gemfile]: Yaaaaayyyy...Sinatra! 🙃

File: app/jobs/rainy_day_texter_job.rb [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+      to: "+1#{user.telephone}",
+      body: "Heads up from thirsty plants! There is a #{chance}% chance of precipitation in your area today."
```
Comment (le3ah) [inline app/jobs/rainy_day_texter_job.rb]: Love this text message.

**Pull request merged**